go/ct/spc/specification.go

```diff
@@ -1787,32 +1787,45 @@ func rulesFor(i instruction) []Rule {
 	return res
 }
 
-func callOp() []Rule {
+func callRulesGenerator() []Rule {
 	res := []Rule{}
+
 	// NOTE: this rule only covers Istanbul case in a coarse-grained way.
 	// Follow-work is required to cover other revisions and situations,
 	// as well as special cases currently covered in the effect function.
-	res = append(res, callOpIstanbul()...)
-	return res
-}
+	//                         revision   warm  destColdCost valueZero
+	callFailEffect := func(s *st.State, addrAccessCost vm.Gas) {
+		FailEffect().Apply(s)
+	}
 
-func callOpIstanbul() []Rule {
-	instruction := instruction{
-		op:        CALL,
-		name:      "_istanbul",
-		staticGas: 700,
-		pops:      7,
-		pushes:    1,
+	revs := []Revision{R07_Istanbul, R09_Berlin}
+	var addressAccessCost vm.Gas
+	for _, rev := range revs {
+		for _, warm := range []bool{true, false} {
+			for _, zeroValue := range []bool{true, false} {
+				for _, static := range []bool{true, false} {
+					if rev == R07_Istanbul {
+						addressAccessCost = 0
+					} else if rev == R09_Berlin && warm {
+						addressAccessCost = 100
+					} else if rev == R09_Berlin && !warm {
+						addressAccessCost = 2600
+					}
```

Comment:
To keep all gas-cost parameters together, I would move this part into the `callOp` instruction, similar to the `staticGas` selection code.